# Generate dataset

In [1]:
from csrc.build import generate_from_audio
from csrc.configurations import DatasetConfig as DC
from utils import check_type, extract_audio, count_class
import decoder

## user configurations

In [2]:
# Kowning: the name of the audio is restrained to (name-lang).xxx, you can define the name as you like but remember to add the  standard language type as lang.
# You can name your subtitle file as your want.
# test_clips = (r"./src/src-test/test-eng.mp4", r"./src/src-test/test.ass")


ab = ("./src/src-american-beauty/american-beauty-eng.ac3", "./src/src-american-beauty/american-beauty-eng.ass") # America beauty 美国丽人
dbc = ("./src/src-dallas-buyers-club/dallas-buyers-club-eng.dts", "./src/src-dallas-buyers-club/dallas-buyers-club-eng.srt") # Dallas buyers club 达拉斯买家俱乐部
fkdsc = ("./src/src-fkdsc/src-fkdsc.wav", "./src/src-fkdsc/src-fkdsc.srt") # 疯狂的石头
identity = ("./src/src-identity/src-identity.ac3", "./src/src-identity/src-identity.ass") # Identity 致命ID
jjh = ("./src/src-jjh/src-jjh.wav", "./src/src-jjh/src-jjh.ssa") # 集结号 
kf = ("./src/src-kong-fu/src-kong-fu.wav", "./src/src-kong-fu/src-kong-fu.ass") # 功夫
lac = ("./src/src-la-confidential/src-la-confidential.ac3", "./src/src-la-confidential/src-la-confidential.ass") # LA confidential 洛城机密 
mi = ("./src/src-mission-impossible-iv/src-mission-impossible-iv.ac3", "./src/src-mission-impossible-iv/src-mission-impossible-iv.ass") # Mission impossible IV 碟中谍4
rzdf = ("./src/src-rzdf/src-rzdf.wav", "./src/src-rzdf/src-rzdf.srt") # 让子弹飞 
sls = ("./src/src-shaolin-soccer/src-shaolin-soccer.ac3", "./src/src-shaolin-soccer/src-shaolin-soccer.ass") # 少林足球 
tdkr = ("./src/src-the-dark-knight-rises/the-dark-knight-rises-eng.mp3", "./src/src-the-dark-knight-rises/the-dark-knight-rises-eng.ass") # The dark knight rises 蝙蝠侠-黑暗骑士崛起
tkoh = ("./src/src-the-kingdom-of-heaven/the-kingdom-of-heaven-eng.ac3", "./src/src-the-kingdom-of-heaven/the-kingdom-of-heaven-eng.ass") # The kingdom of heaven 天国王朝
tks = ("./src/src-the-kings-speech/the-kings-speech-eng.ac3", "./src/src-the-kings-speech/the-kings-speech-eng.ass") # The kings speech 国王的演讲
tmr = ("./src/src-the-matrix-reloaded/src-the-matrix-reloaded.wav", "./src/src-the-matrix-reloaded/src-the-matrix-reloaded.ass") # The matrix reloaded 黑客帝国：重装上阵 
tw = ("./src/src-the-warlords/src-the-warlords.wav", "./src/src-the-warlords/src-the-warlords.ass") # 投名状

dataset_source_long = [ab, dbc, fkdsc, identity, jjh, kf, lac, rzdf, sls, mi, tdkr, tkoh, tks, tmr, tw]

dataset_source_short = [ab, dbc, fkdsc, identity, jjh, kf]

dataset_source_medium = [ab, dbc, fkdsc, identity, jjh, kf, lac, rzdf, sls, mi]

sources = [
    # (audio file path under src, subtitle file path correspondingly under src)\n",
    *dataset_source_long
]

# Destination folder for you to store current sources clips.
# It"s ok if you want to store clips directly under the data folder.
# and this won"t effect whether you want to separate the train/test folder by yourself.

dest_path = ("./data/standard-p2-32khz/")

In [3]:
_, _, _ = count_class(dest_path)


Label 1 instances: 0
Label 0 instances: 0
Total clips: 0



## Start auto generation

In [4]:
for source_file_path, subtitle_file_path in sources: 
    if subtitle_file_path.split(".")[-1] in ["ass", "ssa"]:
        subtitle_decoder = decoder.ASSDecoder
    if subtitle_file_path.split(".")[-1]=="srt":
        subtitle_decoder = decoder.SRTDecoder   
    is_video = check_type(source_file_path)
    audio_file_path = extract_audio(source_file_path, format=DC.dataset_audio_format) if is_video else source_file_path
    generate_from_audio(audio_path=audio_file_path, sub_path=subtitle_file_path, dest_path=dest_path, sub_decoder=subtitle_decoder,  verbose=False)

Extracting timestamps from the subtitle file...
Extraction complete!

Librosa loading audio...
Loading file: ./src/src-american-beauty/american-beauty-eng.ac3
Loading completed. Cost 238.87s

Loading source file success! Using sampling rate 32000.

Start building process.
Transforming dataset...

Building process finished for american-beauty-eng.
Label 1 (speech) clips: 2087
Label 0 (non-speech) clips: 1564 

Extracting timestamps from the subtitle file...
Extraction complete!

Librosa loading audio...
Loading file: ./src/src-dallas-buyers-club/dallas-buyers-club-eng.dts
Loading completed. Cost 234.71s

Loading source file success! Using sampling rate 32000.

Start building process.
Transforming dataset...

Building process finished for dallas-buyers-club-eng.
Label 1 (speech) clips: 2086
Label 0 (non-speech) clips: 1423 

Extracting timestamps from the subtitle file...
Extraction complete!

Librosa loading audio...
Loading file: ./src/src-fkdsc/src-fkdsc.wav
Loading completed. Cost 17

In [5]:
_, _, _ = count_class(dest_path)


Label 1 instances: 30383
Label 0 instances: 26787
Total clips: 57170



### LOG

Building process finished for american-beauty-eng.
Label 1 (speech) clips: 2087
Label 0 (non-speech) clips: 1564 

Building process finished for dallas-buyers-club-eng.
Label 1 (speech) clips: 2086
Label 0 (non-speech) clips: 1423 

Building process finished for src-fkdsc.
Label 1 (speech) clips: 1820
Label 0 (non-speech) clips: 1309 

Building process finished for src-identity.
Label 1 (speech) clips: 1584
Label 0 (non-speech) clips: 1118 

Building process finished for src-jjh.
Label 1 (speech) clips: 2232
Label 0 (non-speech) clips: 1484 

Building process finished for src-kong-fu.
Label 1 (speech) clips: 1124
Label 0 (non-speech) clips: 1859 

Building process finished for src-la-confidential.
Label 1 (speech) clips: 2747
Label 0 (non-speech) clips: 1390 

Building process finished for src-rzdf.
Label 1 (speech) clips: 3124
Label 0 (non-speech) clips: 842 

Building process finished for src-shaolin-soccer.
Label 1 (speech) clips: 1635
Label 0 (non-speech) clips: 1655 

Building process finished for src-mission-impossible-iv.
Label 1 (speech) clips: 1990
Label 0 (non-speech) clips: 1999 

Building process finished for the-dark-knight-rises-eng.
Label 1 (speech) clips: 3046
Label 0 (non-speech) clips: 1891 

Building process finished for the-kingdom-of-heaven-eng.
Label 1 (speech) clips: 2379
Label 0 (non-speech) clips: 3308 

Building process finished for the-kings-speech-eng.
Label 1 (speech) clips: 2419
Label 0 (non-speech) clips: 1136 

Building process finished for src-the-matrix-reloaded.
Label 1 (speech) clips: 441
Label 0 (non-speech) clips: 3707 

Building process finished for src-the-warlords.
Label 1 (speech) clips: 1669
Label 0 (non-speech) clips: 2102 

TOTAL: 
Label 1 instances: 30383
Label 0 instances: 26787
Total clips: 57170
